subjects/java/checkpoints/circular-linkedlist/README.md

```diff
@@ -0,0 +1,118 @@
+## Circular Linked List
+
+### Instructions
+
+Create a circular single linked list data structure that implements the following methods:
+
+- `at(int index)`: to access an element by its index.
+- `add(int value)`: to add an element at the end of the list.
+- `remove(int index)`: to remove an element by its index.
```

Comment:
What should happen if the index is out of bounds? 
What should happen for a negative index? Should `-1` refer to the last item, `-2` to the second last ...?

Reply:
the same as `SingleLinkedList`
calling at with -1/-2: will return -1
calling remove with -1/-2: nothing will be removed the index not found.

it's already defined in the subject.

Reply:
It's not specified as in the `SingleLinkedList` case. 

From `SingleLinkedList` subject:
> - `at(int index)`: to access an element by its index, if the index is out of bound return -1.

Since this is a "circular" linked list, I guess no index is out of bound, but it should be clear that negative index are not accepted

Reply:
It's not specified as in the `SingleLinkedList` case. 

From `SingleLinkedList` subject:
> - `at(int index)`: to access an element by its index, if the index is out of bound return -1.

Since this is a "circular" linked list, I guess no index is out of bounds, but it should be clear that negative indexes are not accepted

Reply:
Oh sorry, my bad i though i put the same description in both yes you are right i will fix that. thank you thank you